decidim-core/app/models/decidim/taxonomy.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Decidim
+  # Represents a hierarchical classification used to organize various entities within the system.
+  # Taxonomies are primarily used to categorize and manage different aspects of participatory spaces,
+  # such as proposals, assemblies, and other components, within an organization.
+  class Taxonomy < ApplicationRecord
+    include Decidim::TranslatableResource
+    include Decidim::FilterableResource
+    include Decidim::Traceable
+
+    after_initialize :set_default_weight
```

Comment:
Is this really a problem?
In your example this happens because you don't save the object to the database, so it's impossible how many of them are initialized to establish a consecutive number. If you save the first one, then the second. You will get the weight ok.

On the other hand, is not a problem to have many taxonomies with the same weight, this is intended to use as a sorting field and really the place to order them is to use the drag & drop feature in the admin editor. If two taxonomies have the same weight they will just get ordered by postgresql, admin will see it, if they don't like it they can just change it.

Although if you have a suggestion on how to do this better, please do!

Reply:
Good, point. Done it.